src/back/hlsl/writer.rs

```diff
@@ -2117,6 +2147,39 @@ impl<'a, W: fmt::Write> super::Writer<'a, W> {
                         let result_ty = func_ctx.info[expr].ty.clone();
                         self.write_storage_load(module, var_handle, result_ty, func_ctx)?;
                     }
+                    // We treat matrices of the form `matCx2` as a sequence of C `vec2`s
+                    // (see top level module docs for details).
+                    Some(crate::AddressSpace::Uniform) => {
+                        if let Expression::GlobalVariable(handle) = func_ctx.expressions[pointer] {
```

Comment:
I think this isn't going to handle accesses to arrays of matrices, like `transform[1]` from my comment above.